oval-schemas/independent-definitions-schema.xsd

```diff
@@ -1309,6 +1321,179 @@
+                                                                  </sch:pattern>
+                                                            </xsd:appinfo>
+                                                      </xsd:annotation>
+                                                </xsd:element>
+                                                <xsd:element name="version" type="oval-def:EntityObjectStringType">
+                                                      <xsd:annotation>
+                                                            <xsd:documentation>The version entity defines the specific version of the database engine to use. This is also important in determining the correct driver to use for establishing a connection.</xsd:documentation>
+                                                            <xsd:appinfo>
+                                                                  <sch:pattern id="ind-def_sql512_object_version">
+                                                                        <sch:rule context="ind-def:sql512_object/ind-def:version">
+                                                                              <sch:assert test="not(@operation) or @operation='equals'"><sch:value-of select="../@id"/> - operation attribute for the version entity of an sql512_object should be 'equals', note that this overrules the general operation attribute validation (i.e. follow this one)</sch:assert>
+                                                                        </sch:rule>
+                                                                  </sch:pattern>
+                                                            </xsd:appinfo>
+                                                      </xsd:annotation>
+                                                </xsd:element>
+                                                <xsd:element name="instance" type="oval-def:EntityObjectStringType">
+                                                      <xsd:annotation>
+                                                            <xsd:documentation>The instance entity defines the specific instance name to be used when connecting to the correct database, where instance refers to the running instance of the DMBS software itself. This could be a separate installation of binaries (such as with MS SQL Server), or just a set of running processes used to manage the DBMS.</xsd:documentation>
+															<xsd:documentation>The OVAL interpreter will automatically determine the list of available instances on the target.</xsd:documentation>
+															<xsd:documentation>When a pattern or string is entered, the OVAL interpeter will consider any matching instance as in scope for analysis.</xsd:documentation>
+													  </xsd:annotation>
+                                                </xsd:element>
+                                                <xsd:element name="database" type="oval-def:EntityObjectStringType" nillable="true">
```

Comment:
So, what does it mean if `xsi:nil="true"`?  (ETA - another comment below)

For that matter, what if the database doesn't have a "Tablespace" type of concept (e.g., Derby DB -- which isn't supported in OVAL, but I happen to know it doesn't have such a concept)?

Reply:
if xsi:nil = 'true', we just run the query against the DBMS instance, not a single specific database.  I was debating the whole 'default' database terminology as it was confusing to me as well.  Sometimes these queries are against a specific 'master' database, but other queries are are very bizzare, and may be SQL Server centric, but maybe other DBMS's have similar things, see below:

SELECT
    SERVERPROPERTY('IsClustered') AS [is_clustered],
    SERVERPROPERTY('IsHadrEnabled') AS [is_hadr_enabled]

This queries some configuration setting for the SQL Server instance, and is not targeted at any single database table.